kinesis-video-pic/src/utils/src/Mutex.c

```diff
@@ -217,24 +205,21 @@ VOID defaultFreeMutex(MUTEX mutex)
     pthread_mutex_t* pMutex = (pthread_mutex_t*) mutex;
     pthread_mutex_destroy(pMutex);
 
-    // De-allocate the memory if it's not a well-known mutex - aka if we had allocated it previously
-    if (pMutex != &globalKvsReentrantMutex && pMutex != &globalKvsNonReentrantMutex) {
-        MEMFREE(pMutex);
-    }
+    MEMFREE(pMutex);
 }
 
-pthread_cond_t globalKvsConditionVariable = PTHREAD_COND_INITIALIZER;
-
 CVAR defaultConditionVariableCreate()
 {
     CVAR pVar = (CVAR)MEMCALLOC(1, SIZEOF(pthread_cond_t));
-    if (NULL == pVar) {
-        return &globalKvsConditionVariable;
-    }
+    pthread_condattr_t attr;
 
-    if (0 != pthread_cond_init(pVar, NULL)) {
-        return &globalKvsConditionVariable;
-    }
+    pthread_condattr_init (&attr);
+#if defined __linux__ || defined __CYGWIN__
```

Comment:
Do we need this condition? This seems to be a portable API

Reply:
macosx does not like it.

Reply:
Indeed it doesn't